packages/angular-workspace/nimble-angular/src/thirdparty/directives/tests/router_integration.spec.ts

```diff
@@ -7146,7 +7146,9 @@ class DummyLinkCmp {
 class AbsoluteSimpleLinkCmp {
 }
 
-@Component({selector: 'link-cmp', template: `<a [routerLink]="['../simple']">link</a>`})
+// [Nimble] add irrelevant :not(body) to selector to force distinct component ID generation
+//  Solution based on suggestion from https://angular.io/errors/NG0912#debugging-the-error
+@Component({selector: 'link-cmp:not(body)', template: `<a [routerLink]="['../simple']">link</a>`})
```

Comment:
instead of whack-a-mole can we uniqify all the `selector: 'link-cmp'` selectors?
choose a pattern that's less random elements, maybe `link-cmp:not([nimble-unused-0])`, `link-cmp:not([nimble-unused-1])`, etc

Reply:
I think the rest of the components with that selector have [different enough template characteristics](https://github.com/ni/nimble/pull/2087#discussion_r1596098105) that they generate different IDs (note that we've only ever seen collisions involving `RelativeLinkCmp` and/or `AbsoluteSimpleLinkCmp`), but it can't hurt to play it safe.